packages/async-rewriter2/src/async-writer-babel.ts

```diff
@@ -521,17 +566,46 @@ export const makeMaybeAsyncFunctionPlugin = ({ types: t }: { types: typeof Babel
           }
 
           // Transform expression `foo` into
-          // `(ex = foo, isSyntheticPromise(ex) ? await ex : ex)`
+          // `('\uFEFFfoo\uFEFF', ex = foo, isSyntheticPromise(ex) ? await ex : ex)`
+          // The first part of the sequence expression is used to identify this
+          // expression for re-writing error messages, so that we can transform
+          // TypeError: ((intermediate value)(intermediate value) , (intermediate value)(intermediate value)(intermediate value)).findx is not a function
+          // back into
+          // TypeError: db.test.findx is not a function
+          // The U+FEFF markers are only used to rule out any practical chance of
+          // user code accidentally being recognized as the original source code.
           const { expressionHolder, isSyntheticPromise } = identifierGroup;
+          const originalSource = t.stringLiteral(
+            '\ufeff' + (this.file as any).code.slice(path.node.start, path.node.end) + '\ufeff');
           path.replaceWith(Object.assign(
             awaitSyntheticPromiseTemplate({
+              ORIGINAL_SOURCE: originalSource,
               EXPRESSION_HOLDER: expressionHolder,
               ISP_IDENTIFIER: isSyntheticPromise,
               NODE: path.node
             }),
             { [isGeneratedHelper]: true }
           ));
         }
+      },
+      CatchClause: {
+        exit(path) {
+          if (path.node[isGeneratedHelper] || !path.node.param || path.node.param.type !== 'Identifier') return;
+          const existingIdentifiers: AsyncFunctionIdentifiers | null =
+            path.findParent(path => !!path.getData(identifierGroupKey))?.getData(identifierGroupKey);
+          if (!existingIdentifiers) return;
+          // Turn `... catch (err) { ... }` into `... catch (err) { err = demangleError(err); ... }`
+          path.replaceWith(Object.assign(
+            t.catchClause(path.node.param,
+              t.blockStatement([
+                t.expressionStatement(
+                  t.assignmentExpression('=', path.node.param,
+                    t.callExpression(existingIdentifiers.demangleError, [path.node.param]))),
+                path.node.body
+              ])),
+            { [isGeneratedHelper]: 1 }
```

Comment:
Is there a reason to have `1` here instead of `true` as in the other places? :)

Reply:
No, it’s just me being odd here :D Will fix